README.md

```diff
@@ -919,6 +919,42 @@ parsers, it may be more convenient to use the `xml_mode` context manager:
     # ... We have now returned to using the default XML parser.
 ```
 
+Application Cleanup
+-------------------
+
+A user-defined function may be registered to be invoked before, during, or after the time
+when the Python iRODS Client is engaged in object teardown (also called "cleanup") operations.
+This is analogous to the atexit facility in C or in Python itself, except that we have the extra
+resolution to specify that a function or callable object be run either just before, or just after, the
+aforementioned object teardown stage:
+
+```
+    from irods import at_client_exit
+    at_client_exit.register_for_execution_after_prc_cleanup(lambda : print("PRC cleanup has completed."))
+    at_client_exit.register_for_execution_before_prc_cleanup(lambda : print("PRC cleanup is about to start."))
+```
+Alternatively, we can use a lower-level interface to access the "during" aspect of things and run a function closer in
+time to the actual Python client cleanup processing.
+```
+    at_client_exit._register(LibraryCleanupStage.DURING, lambda : print("This is printed during the PRC cleanup stage."))
+```
```

Comment:
I'm not sure that we want to advertise using the private-ish function `_register` (leading underscore indicates that... use is discouraged, right?). Perhaps we omit this from the README?

Reply:
Agree.   No public talk about internal thing please.